Gives spoons `TOOL_MINING`, adds spoons to autolathe and prison vending machines

TemporalOroboros opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
- Implements `/obj/item/kitchen/spoon`
- Gives spoons `TOOL_MINING`

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
- This is one of the most basic prison memes, why do we not have this

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: You can dig through rock and dirt with a spoon, elbow grease, and a <i>lot</i> of time
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TemporalOroboros committed:
Makes `/obj/item/kitchen/spoon` not an abstract class

- Implements `/obj/item/kitchen/spoon`
- Gives spoons `TOOL_MINING`

Qustinnus commented:
now add a super long tunnel with just rocks from brig to armory

D4C-420 commented:
Put these in everyone's internals boxes on icebox. Would make it suck less to fall down a z level or two on ice box.

Qustinnus commented:
lol just grab a spoon

D4C-420 commented:
if your going out of your way to grab a spoon why not grab a plasma cutter from cargo? But yeah fair enough.

justinbooj9 commented:
contractible mining spoon now available in your emergency internals box

ArcaneMusic commented:
Plastic Spoons are printable at all departmental lathes, as well as autolathes.

D4C-420 commented:
you could say the same thing about internals. Emergency oxy tanks, and face masks are abundant in the station along with gas mask and full oxy tanks but we still give everyone internals why? Because dying to a single hull breach or n2o canister opening is not fun and having large parts of the station rendered uninhabitable because you cant completely remove the gas is also not fun. Getting rid of everyone's internals boxes would just make people steal all the other ones see #54689 for why people argued it was not merged. Giving everyone a plastic spoon in there internals boxes would fix iceboxes largest problem of people just being stuck in walls and pretty much round removed because a single decent size bomb went off rending parts of a hall way a death trap. Sure they would still have to make their way back to the station but now they at least have a chance instead of being stuck in a wall unable to do anything and just ssding because they did not go-ahead and bitch to chem to make plastic just so they can have a small, slow mining tool in case they need it.

Mothblocks reviewed (dismissed):
emphasis on your environment babyyyyy

Jared-Fogle commented on code/game/objects/items/kitchen.dm:
```diff
@@ -276,15 +276,33 @@
+	toolspeed = 25 // Literally 25 times worse than the base pickaxe
+
 /obj/item/kitchen/spoon/plastic
 	name = "plastic spoon"
 	desc = "Just be careful your food doesn't melt the spoon first."
 	icon_state = "plastic_spoon"
```
remove some of these since they're duped now

TemporalOroboros replied:
Done

Jared-Fogle commented on code/game/objects/items/kitchen.dm:
```diff
@@ -276,15 +276,33 @@
+	attack_verb_continuous = list("whacks", "spoons", "taps")
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 30)
+	custom_materials = list(/datum/material/iron=120)
+	custom_price = PAYCHECK_PRISONER * 5
```
this has a price, but is it in the vending machines?

TemporalOroboros replied:
Now it is

ilikesaying commented:
**make digging with your fingers a last resort, taking more time, elbow grease and damage to arms**

TemporalOroboros committed:
Makes Spoons Accessible

- Adds the spoon design to the autolathe and service lathes
- Adds the spoon to the Dinnerware and Sustenance (Contraband) vendors
- Adds the plastic spoon to the Sustenance vendor
